### PR TITLE
fix: avoid picking up class/className in icon color classes

### PR DESCRIPTION
**Pull request by elevatebart**
## Before

 `class` attribute (or `className` in React) would be wrongly picked up to build icon classes

before

# After

We use the white list of color attributes to filter the color props.

after

### CODE REVIEW

File: icon-registry/build-icons.mjs [hunk 1]
```diff
@@ -199,6 +202,16 @@ async function generateIndex(iconsObjectUnique) {
     dedent`
   // THIS FILE IS AUTO GENERATED BY build-icons.mjs
 
+  /**
+   * All possible prop names for icon colors
+   */
+  export const ICON_COLOR_PROP_MANES = ${JSON.stringify(
```
Comment (mapsandapps) [inline icon-registry/build-icons.mjs:208]: typo?

Comment (elevatebart) [reply icon-registry/build-icons.mjs:208]: What do you mean?

Comment (mapsandapps) [reply icon-registry/build-icons.mjs:208]: `MANES`?

Comment (elevatebart) [reply icon-registry/build-icons.mjs:208]: wow, I need more sleep.
```suggestion
  export const ICON_COLOR_PROP_NAMES = ${JSON.stringify(
```

Comment (elevatebart) [reply icon-registry/build-icons.mjs]: typo fixed

**Comment by mapsandapps** [comment]
`strokeColor` is dark gray for all icons in [the story](https://cypress-design-h7hjqr4kk-cypress-io.vercel.app/?path=/docs/icons--page), regardless of the dropdown status. other uses of icons seem to be fine, though

**Comment by elevatebart** [comment]
@mapsandapps good catch, I renamed a bunch of variables to get their names right but forgot one.

It should be working fine now.

**Review by mapsandapps — APPROVED** [review]

**Pull request merged**